js/components/main.js

```diff
@@ -0,0 +1,55 @@
+var request = new XMLHttpRequest();
+request.onload = onDataLoadedListener;
+request.open("get", "data.json", true);
+request.send();
+
+var data;
+function onDataLoadedListener() {
+    data = JSON.parse(this.responseText);
+    initializeComponent();
+}
+
+function initializeComponent() {
+    var Button = React.createClass({
```

Comment:
each component should be separate file, also better prefer es2015 standart, with `babel` convert, can use `webpack` as a bundle file creator